app/models/lookup_value.rb

```diff
@@ -98,6 +98,18 @@ def ensure_matcher(match_type)
     matcher
   end
 
+  def ensure_matcher_exists
+    return false if match.blank?
+    match.split(LookupKey::KEY_DELM).each do |m|
+      key_element = m.split(LookupKey::EQ_DELM).first
+      unless lookup_key.path.include?(key_element)
```

Comment:
You're right. that's better. The fact that I missed a few or the single quoted strings proves this is still permitting unknown matchers, I'll change it.